mlir/lib/Dialect/SPIRV/IR/SPIRVTypes.cpp

```diff
@@ -171,9 +173,17 @@ void CompositeType::getCapabilities(
       .Case<VectorType>([&](VectorType type) {
         auto vecSize = getNumElements();
         if (vecSize == 8 || vecSize == 16) {
-          static const Capability caps[] = {Capability::Vector16};
-          ArrayRef<Capability> ref(caps, std::size(caps));
-          capabilities.push_back(ref);
+          static constexpr Capability caps[] = {Capability::Vector16,
+                                                Capability::VectorAnyINTEL};
+          capabilities.push_back(caps);
+        }
+        // VectorAnyINTEL capability removes the vector size restriction and
```

Comment:
Not sure I follow the logic here. We are pushing duplicated capabilities here? Shouldn't we do the check `vecSize > 4` to include `VectorAnyINTEL` first, and then additionally add `Vector16` if `vecSize == 8 || vecSize == 16`?